### PR TITLE
Move actions from client to report

### PR DESCRIPTION
**Pull request by esjee**
fixes https://github.com/oreoshake/hackerone-client/issues/20

* I had to change some of the VCRs to change the report id
* I split up the 'add_comment' VCR yml into 'add_comment' (internal) and 'add_public_comment' (not internal)
* I removed the test to see if the methods 404 when provided an invalid report id, since the report id argument is now inferred from the report object itself

### CODE REVIEW

File: lib/hackerone/client.rb [hunk 1]
```diff
@@ -225,7 +125,12 @@ def self.parse_response(response)
         elsif response.status.to_s.start_with?("5")
           raise RuntimeError, "API called failed, probably their fault: #{response.body}"
         elsif response.success?
-          JSON.parse(response.body, :symbolize_names => true)[:data]
+          response_body_json = JSON.parse(response.body, :symbolize_names => true)
```
Comment (oreoshake) [inline lib/hackerone/client.rb]: Interesting. Did this fix a bug?

Comment (esjee) [reply lib/hackerone/client.rb]: I think so; I didn't verify in the old code but when changing a reference the response doesn't contain `data` as a top-level key. You can see that here:

```
@report = response_json[:relationships][:report][:data]
```

File: lib/hackerone/client/report.rb [hunk 2]
```diff
@@ -126,6 +147,87 @@ def suggest_bounty(message:, amount:, bonus_amount: nil)
...
+        add_report_reference(reference)
+        state_change(:triaged)
+      end
+
+      # Add a comment to a report. By default, internal comments will be added.
+      #
+      # id: the ID of the report
+      # message: the content of the comment that will be created
+      # internal: "team only" comment (true, default) or "all participants"
+      def add_comment(message, internal: true)
```
Comment (oreoshake) [inline lib/hackerone/client/report.rb]: What about having this return a `HackerOne::Client::Activities::CommentAdded`?

File: lib/hackerone/client/report.rb [hunk 3]
```diff
@@ -126,6 +147,87 @@ def suggest_bounty(message:, amount:, bonus_amount: nil)
         Activities.build(response_body)
       end
 
+      ## Idempotent: change the state of a report. See STATES for valid values.
+      #
+      # id: the ID of the report
+      # state: the state in which the report is to be put in
+      #
+      # returns an HackerOne::Client::Report object or raises an error if
```
Comment (oreoshake) [inline lib/hackerone/client/report.rb]: I think this returns a hash (but should return an `Report`)

File: lib/hackerone/client/report.rb [hunk 4]
```diff
@@ -126,6 +147,87 @@ def suggest_bounty(message:, amount:, bonus_amount: nil)
...
+        body = {
+          type: "issue-tracker-reference-id",
+          attributes: {
+            reference: reference
+          }
+        }
+
+        response_json = make_post_request("reports/#{id}/issue_tracker_reference_id", request_body: body)
+        @report = response_json[:relationships][:report][:data]
+        self
```
Comment (oreoshake) [inline lib/hackerone/client/report.rb]: :+1:

File: lib/hackerone/client/report.rb (hunk 2 above)
Comment (esjee) [reply lib/hackerone/client/report.rb]: That's a good idea!

File: lib/hackerone/client/report.rb (hunk 3 above)
Comment (esjee) [reply lib/hackerone/client/report.rb]: You're right, I broke the docstring.

**Comment by esjee** [comment]
Thanks for the comments @oreoshake !
I've tried to address them, let me know what you think!

**Comment by oreoshake** [comment]
Released in 0.9.0

**Pull request merged**